### PR TITLE
[ONNX] Use parameter values in onnx shape inference

### PR DESCRIPTION
**Pull request by BowenBao**
Adds an additional run of onnx shape inference after constant folding, since initializer may have changed and affected shape inference.

### CODE REVIEW

File: torch/csrc/jit/passes/onnx/shape_type_inference.cpp [hunk 1]
```diff
@@ -526,8 +548,13 @@ void ONNXAssignOutputShape(
   }
 }
 
-void ONNXShapeTypeInference(std::shared_ptr<Graph>& graph, int opset_version) {
-  ONNXShapeTypeInference(graph->block(), opset_version);
+void ONNXShapeTypeInference(
+    std::shared_ptr<Graph>& graph,
+    const ParamMap& params_dict,
+    int opset_version) {
+  for (auto n : graph->nodes()) {
```
Comment (neginraoof) [inline torch/csrc/jit/passes/onnx/shape_type_inference.cpp:555]: Why changing block inference to node inference?

File: torch/csrc/jit/passes/onnx/shape_type_inference.cpp [hunk 2]
```diff
@@ -255,34 +255,50 @@ Value* CloneValueFromListConstruct(Value* v, std::shared_ptr<Graph> n_graph) {
...
+          case ::c10::prim::ListConstruct: {
+            return CloneValueFromListConstruct(v, n_graph);
+          }
+          case ::c10::prim::PackPadded: {
+            auto input = n_graph->addInput();
+            input->copyMetadata(v_n->input(0));
+            return input;
+          }
+          default: {
+            if (vals_to_params_map.find(v) != vals_to_params_map.end()) {
```
Comment (neginraoof) [inline torch/csrc/jit/passes/onnx/shape_type_inference.cpp:283]: Changes in this part look good to me.
Just a note that this expects parameters to be tensors.

File: torch/csrc/jit/passes/onnx/shape_type_inference.cpp (hunk 1 above)
Comment (BowenBao) [reply torch/csrc/jit/passes/onnx/shape_type_inference.cpp]: thanks for catching, looks like out dated merge, reverting to calling block version.

**Review by neginraoof — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**